File: shorthand/lib/shorthand.nim
import nimpy
import os
import strformat, strutils
import configparser

proc short() {.exportpy.} =
    var cfg = parseIni(readFile("BP/filters/shorthand/config.ini"))
    var shorts = cfg.getProperty("define", "shorts")
    var pshort = shorts.split(" ")
    for file in os.walkFiles(fmt"BP/functions/*.mcfunction"):
        var rFile = readFile(file)
        var rNum = -1
        for i in pshort:
            rNum.inc(1)
            var short = pshort[rNum]
            var long = cfg.getProperty("as", fmt"{pshort[rNum]}_long")
            echo fmt"short = {short} long = {long}"
            rFile = rFile.replace(short, long)
        os.removeFile(file)
        writeFile(file, rFile)

File: shorthand/shorthand.py
import configparser
import os
import glob

config = configparser.ConfigParser()
if not os.path.isdir("data/shorthand"):
    os.mkdir("data/shorthand")
if not os.path.isfile("data/shorthand/config.ini"):
    with open('data/shorthand/config.ini', 'x') as f:
        f.write("[define]\nshorts = admin armors\n[as]\nadmin_long = @a[tag=admin]\narmors_long = @e[type=armor_stand]")
cfg = config.read('data/shorthand/config.ini')
shorts = cfg["define"]["shorts"]
pshort = shorts.split()
for filename in glob.iglob("./BP/functions/" + '**/*.mcfunction', recursive=True):
    print(filename)
    ropen = open(filename, "r")
    rFile = ropen.read()
    ropen.close()
    for i in pshort:
        short = i
        long = cfg["as"]["{}_long".format(i)]
        print("short = {arg1}, long = {arg2}".format(arg1 = short, arg2 = long))
        rFile = rFile.replace(short, long)
    remove(filename)
    newopen = open(filename, "x")
    newopen.write(rFile)
    newopen.close()
